activerecord/test/cases/insert_all_test.rb

```diff
@@ -225,6 +226,23 @@ def test_upsert_all_updates_existing_records
     assert_equal new_name, Book.find(1).name
   end
 
+  def test_upsert_all_updates_existing_record_by_primary_key
+    skip unless supports_insert_on_duplicate_update?
+
+    new_name = "Agile Web Development with Rails, 4th Edition"
+    Book.upsert_all [{ id: 1, name: new_name }], unique_by: :id
+    assert_equal new_name, Book.find(1).name
```

Comment:
```suggestion
    Book.upsert_all [{ id: 1, name: "New edition" }], unique_by: :id

    assert_equal "New edition", Book.find(1).name
```

I'd rather we just shorten the text and repeat it than capture it in a local variable.